pkg/osquery/tables/filevault/filevault.go

```diff
@@ -0,0 +1,56 @@
+// +build darwin
+
+package filevault
+
+import (
+	"bytes"
+	"context"
+	"os/exec"
+	"strings"
+
+	"github.com/go-kit/kit/log"
+	"github.com/kolide/osquery-go"
+	"github.com/kolide/osquery-go/plugin/table"
+	"github.com/pkg/errors"
+)
+
+const fdesetupPath = "/usr/bin/fdesetup"
+
+type Table struct {
+	client *osquery.ExtensionManagerClient
+	logger log.Logger
+}
+
+func TablePlugin(client *osquery.ExtensionManagerClient, logger log.Logger) *table.Plugin {
+	columns := []table.ColumnDefinition{
+		table.TextColumn("status"),
+	}
+
+	t := &Table{
+		client: client,
+		logger: logger,
+	}
+
+	return table.NewPlugin("kolide_filevault", columns, t.generate)
+}
+
+func (t *Table) generate(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+	var results []map[string]string
+
+	// Read the system's fdesetup configuration
+	var stdout bytes.Buffer
+	stdout.Reset()
+	cmd := exec.CommandContext(ctx, fdesetupPath, "status")
+	cmd.Stdout = &stdout
+	if err := cmd.Run(); err != nil {
+		return nil, errors.Wrap(err, "calling fdesetup")
+	}
+	output := string(stdout.Bytes())
+	status := strings.TrimSuffix(output, "\n")
+	result := map[string]string{
+		"status": status,
+	}
+	results = append(results, result)
```

Comment:
This is fine, but since this is a single line you could just return it directly. (FYI)

Reply:
Gonna keep this as is, not familiar enough with golang syntax to do this a different way.